### ISSUE TITLE
CSS Modules hashes clash within the same file

### ISSUE THREAD

**Issue by markdalgleish** [parcel-bundler/lightningcss#351]
I ran into an issue where styles worked in dev but broke in the production build because I set the `pattern` option to `[hash]` but it turned out that the hashes were the same for each `.module.css` file.

You can see this in the current test suite. Notice how the classes are only unique because of the trailing `_[local]` value. This would break if only the hashes were included: https://github.com/parcel-bundler/lightningcss/blob/27cb9c2a382fd8d9702de8def9dd18db54280c43/tests/cli_integration_tests.rs#L95-L129

**Comment by devongovett** [comment]
At the moment `[hash]` is just a hash of the file path. I guess we could add a new pattern option to hash the whole class, or maybe change the behavior, but I avoided this initially for performance reasons.

**Comment by markdalgleish** [comment]
Could the hash include a counter at the end? This is what we do in Vanilla Extract, so in this example we'd end up with `EgL3uq0`, `EgL3uq1` etc.

**Comment by markdalgleish** [comment]
Ah, that's interesting. It's also a bit of a gotcha that the build suddenly breaks if I start using grid line names when I have a pattern that doesn't end in `[local]`.

**Comment by devongovett** [comment]
Yeah, might add some validation of the config to ensure that it will generate unique, valid names.

The counter at the end is an interesting idea, but might not be stable over time e.g. if you insert a new rule. Will continue to think through options.

**Comment by sciyoshi** [comment]
Ran into the same surprise. #1237 should fix this with an opt-in `Config::hash_local_name` flag — when set, the local class name is mixed into the hash input so classes within the same file get distinct hashes. Off by default to preserve existing output, but worth a discussion about whether to flip the default in a future major to prevent this footgun.
